pkg/apply/taskrunner/runner.go

```diff
@@ -71,10 +71,10 @@ func (tsr *taskStatusRunner) Run(ctx context.Context, taskQueue chan Task,
 
 // NewTaskRunner returns a new taskRunner. It can process taskqueues
 // that does not contain any wait tasks.
+// TODO: Do we need this abstraction layer now that baseRunner doesn't need a collector?
```

Comment:
Probably not. I'm ok with simplifying things here.

Reply:
I think I'll leave that for a follow up PR. This one's already big enough.